python/triton/language/semantic.py

```diff
@@ -1319,6 +1319,9 @@ def _str_to_dot_input_precision(input_precision, builder):
 def dot(lhs: tl.tensor, rhs: tl.tensor, acc: tl.tensor, input_precision: Optional[str], max_num_imprecise_acc: int,
         out_dtype: tl.dtype, builder: ir.builder) -> tl.tensor:
 
+    def support_m16n16k8():
```

Comment:
For fp8 and int8 on MI300, the mfma instructions are `32X32X16` and `16X16X32` which is not applicable here

Reply:
I added a check on the architecture. This will be always a bit fragile until we tackle the general problem of supporting any tl.dot size (and let the compiler decide how to accelerate it)

Reply:
from https://github.com/triton-lang/triton/blob/main/lib/Analysis/Utility.cpp#L418, this only effectively enables 32x32x8 as of right now for AMD backend?